### DISCUSSION TITLE
Still Reachable Leaks Found in Ulfius

### DISCUSSION THREAD

**Discussion by gayathriaccount** [babelouest/ulfius#213] [Q&A] [answered]
Hi,

When testing Ulfius framework for REST Interface support in our project, few still reachable leaks were encountered in the below scenario.

  **Scenario:**
    Client posts a HTTP request to the server.

   To make sure the API usages were proper, the test was performed using the example code provided and we could find leaks there as well.
   The same APIs as present in the example code is used in our project for testing this scenario.

Please find the attached leak report
[log_soap_client.txt](https://github.com/babelouest/ulfius/files/7779696/log_soap_client.txt)

Could you please let us know if there are any APIs which could be used to resolve these leaks.

Thanks and regards
Gayathri

**Comment by babelouest** [comment, marked as answer]
Hello,

A `still reachable` leak is usually not a lost allocated memory. It can be, for example, blocks of memory allocated by a library, and kept in memory to be reused by other instances that use the same library.

https://valgrind.org/docs/manual/faq.html#faq.deflost
https://stackoverflow.com/questions/3840582/still-reachable-leak-detected-by-valgrind/3857638#3857638
